addons/sale_timesheet/models/project.py

```diff
@@ -315,51 +336,32 @@ def get_panel_data(self):
-        }
-        product_uom_unit = self.env.ref('uom.product_uom_unit')
-        for sol in sols:
-            name = [x[1] for x in sol.name_get()] if number_sale_orders > 1 else sol.name
-            qty_delivered = sol.product_uom._compute_quantity(sol.qty_delivered, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
-            product_uom_qty = sol.product_uom._compute_quantity(sol.product_uom_qty, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
-            if qty_delivered > 0 or product_uom_qty > 0:
-                sold_items['data'].append({
-                    'name': name,
-                    'value': '%s / %s %s' % (formatLang(self.env, qty_delivered, 1), formatLang(self.env, product_uom_qty, 1), sol.product_uom.name if sol.product_uom == product_uom_unit else self.env.company.timesheet_encode_uom_id.name),
-                    'color': 'red' if qty_delivered > product_uom_qty else 'black'
-                })
-                #We only want to consider hours and days for this calculation, and eventually units if the service policy is not based on milestones
-                if sol.product_uom.category_id == self.env.company.timesheet_encode_uom_id.category_id or (sol.product_uom == product_uom_unit and sol.product_id.service_policy != 'delivered_manual'):
-                    sold_items['total_sold'] += product_uom_qty
-                    sold_items['effective_sold'] += qty_delivered
-        remaining = sold_items['total_sold'] - sold_items['effective_sold']
-        sold_items['remaining'] = {
-            'value': remaining,
-            'color': 'red' if remaining < 0 else 'black',
-        }
-        return sold_items
-
     def _get_sale_order_items_query(self, domain_per_model=None):
         if domain_per_model is None:
-            domain_per_model = {}
+            domain_per_model = {'project.task': [('allow_billable', '=', True)]}
+        else:
+            domain_per_model['project.task'] = expression.AND([
+                domain_per_model.get('project.task', []),
+                [('allow_billable', '=', True)],
+            ])
         query = super()._get_sale_order_items_query(domain_per_model)
+
+        Timesheet = self.env['account.analytic.line']
+        timesheet_domain = [('project_id', 'in', self.ids), ('so_line', '!=', False), ('project_id.allow_billable', '=', True)]
+        if Timesheet._name in domain_per_model:
+            timesheet_domain = expression.AND([
+                domain_per_model.get(Timesheet._name, []),
+                timesheet_domain,
+            ])
+        timesheet_query = Timesheet._where_calc(timesheet_domain)
+        Timesheet._apply_ir_rules(timesheet_query, 'read')
+        timesheet_query_str, timesheet_params = timesheet_query.select(
+            f'{Timesheet._table}.project_id AS id',
+            f'{Timesheet._table}.so_line AS sale_line_id',
+        )
```

Comment:
That's actually what ``_search`` does. 

Couldn't you retrieve your query_str and params from the result query when calling it ?

Reply:
Yeah, I tried to do it, but I have an error saying the final query is not valid. It is probably because of the `UNION` that I did. I guess it is because `_search` also add `ORDER BY` 🤔
I will investigate it.